sdk/storage/inc/datalake/path_client.hpp

```diff
@@ -207,18 +226,71 @@ namespace Azure { namespace Storage { namespace DataLake {
     /**
      * @brief Sets the properties of a resource the path points to.
      * @param options Optional parameters to set the properties to the resource the path points to.
-     * @return PathUpdateResponse
+     * @return SetPropertiesResult
      */
-    PathUpdateResponse SetProperties(
+    SetPropertiesResult SetProperties(
         const SetPathPropertiesOptions& options = SetPathPropertiesOptions()) const;
 
+    /**
+     * @brief Get Properties returns all system and user defined properties for a path. Get Status
+     *        returns all system defined properties for a path. Get Access Control List returns the
+     *        access control list for a path.
+     * @param options Optional parameters to get the properties from the resource the path points
+     *                to.
+     * @return GetPathPropertiesResponse
+     */
+    GetPathPropertiesResponse GetProperties(
+        const PathGetPropertiesOptions& options = PathGetPropertiesOptions()) const;
+
+    /**
+     * @brief Sets the metadata of a resource the path points to.
+     * @param options Optional parameters to set the metadata to the resource the path points to.
+     * @return SetMetadataResult
+     */
+    SetMetadataResult SetMetadata(
+        const std::map<std::string, std::string> metadata,
+        const SetPathMetadataOptions& options = SetPathMetadataOptions()) const;
+
+    /**
+     * @brief Get the metadata for a path.
+     * @param options Optional parameters to get the metadata from the resource the path points to.
+     * @return std::map<std::string, std::string>
+     */
+    std::map<std::string, std::string> GetMetadata(
+        const GetPathMetadataOptions& options = GetPathMetadataOptions()) const;
+
     /**
      * @brief Creates a file or directory. By default, the destination is overwritten and
      *        if the destination already exists and has a lease the lease is broken.
      * @param options Optional parameters to create the resource the path points to.
      * @return PathCreateResponse
      */
-    PathCreateResponse Create(const PathCreateOptions& options = PathCreateOptions()) const;
+    PathCreateResponse Create(
+        const PathResourceType& type,
+        const PathCreateOptions& options = PathCreateOptions()) const;
+
+    /**
+     * @brief Creates as a file. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param options Optional parameters to create the resource the path points to.
+     * @return PathCreateResponse
+     */
+    PathCreateResponse CreateAsFile(const PathCreateOptions& options = PathCreateOptions()) const
```

Comment:
How do they name these two API in other language sdk?

Reply:
`CreateFile` or `CreateDirectory`. However other languages don't have the issue of name clashing with what is defined in `fileapi.h` :(

Reply:
Does compiler complain about clashes?

Things in `fileapi.h` are C API. We're writing C++ functions, inside namespace. I think that's why namespace was invented.

Reply:
Well, you are actually right. However, VS is having some issues doing the IntelliSense when I changed the name to CreateFile/CreateDirectory, tricking me into believing the name to introduce an issue. Changed in a newer iteration.